pkg/upgrade/upgrade.go

```diff
@@ -0,0 +1,246 @@
+package upgrade
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/mitchellh/go-ps"
+	"github.com/opencontainers/selinux/go-selinux"
+	"github.com/rancher/k3s-upgrade/pkg/types"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func fileExists(path string) bool {
```

Comment:
This function seems unnecessary. Most if not all of the calls into it should either be skipping the check or making it explicitly on their own (and returning the non "file not found" error conditions to the calling code).